File: javascript/triangle/triangle.js
export class Triangle {
    #sides
    #isReal
    constructor(...sides) {
      this.#sides = sides;
      const [a, b, c] = sides.sort((a, b) => a - b)
      this.#isReal = (a + b > c) && sides.length === 3
    }

    get isEquilateral() {
      const [a, b, c] = this.#sides
      return [a, b, c].every((v, i, a) => 
          (v === a[0] && v!== 0 && v != null)
      ) && this.#isReal
    }

    get isIsosceles() {
      const [a, b, c] = this.#sides
      return ((a == b || b == c || a == c)) && this.#isReal
    }

    get isScalene() {
      const [a, b, c] = this.#sides
      return (a != b && b != c && a != c) && this.#isReal
    }
}

File: c/triangle/triangle.c
#include "triangle.h"

static bool is_valid(triangle_t s){
  return s.a && s.b && s.c && (s.a + s.b >= s.c) && (s.a + s.c >= s.b) 
         && (s.b + s.c >= s.a);
}

bool is_equilateral(triangle_t sides){
  return is_valid(sides) && (sides.a == sides.b) && (sides.a == sides.c);
}

bool is_isosceles(triangle_t sides){
  return is_valid(sides) && ((sides.a == sides.b) || (sides.a == sides.c) 
                             || (sides.b == sides.c));
}

bool is_scalene(triangle_t sides){
  return is_valid(sides) && !is_isosceles(sides);
}
